src/datanodes/data-node-1disk-resources.json

```diff
@@ -33,51 +33,11 @@
   "variables": {
     "namespace": "[parameters('vm').namespace]",
     "avSetCount": "[add(div(sub(parameters('vm').count, 1), 100), 1)]",
-    "nodesPerStorageAccount": "[parameters('storageSettings').nodesPerStorageAccount]",
-    "storageAccountPrefix": [
-      "0",
-      "6",
-      "c",
-      "i",
-      "o",
-      "u",
-      "1",
-      "7",
-      "d",
-      "j",
-      "p",
-      "v",
-      "2",
-      "8",
-      "e",
-      "k",
-      "q",
-      "w",
-      "3",
-      "9",
-      "f",
-      "l",
-      "r",
-      "x",
-      "4",
-      "a",
-      "g",
-      "m",
-      "s",
-      "y",
-      "5",
-      "b",
-      "h",
-      "n",
-      "t",
-      "z"
-    ],
-    "storageAccountPrefixCount": "[length(variables('storageAccountPrefix'))]",
-    "newStorageAccountNamePrefix": "[uniqueString(resourceGroup().id, deployment().name)]"
+    "diskCount": 1
```

Comment:
It looks like all `data-node-Ndisk-resources.json` resource files can be refactored into one `data-node-disk-resources.json` file that accepts a `diskCount` parameter

Reply:
Because of we have a 0 datadisk configuration, the diskCount = 0 will cause the copy sentence error. 
So I tried to make minimal changes to the origin as possible.
The completely refactoring of this data node template is not clear in my mind, i think we could more discussing about this.

Reply:
OK, sounds like there would need to be two templates, one for 0 disks and another for N disks.